### PR TITLE
8312489: Increase jdk.jar.maxSignatureFileSize default which is too low for JARs such as WhiteSource/Mend unified agent jar

### PR DESCRIPTION
**Pull request by haimaychao**
JDK-8312489: Increase jdk.jar.maxSignatureFileSize default which is too low for JARs such as WhiteSource/Mend unified agent jar

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change requires CSR request [JDK-8313216](https://bugs.openjdk.org/browse/JDK-8313216) to be approved
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8312489](https://bugs.openjdk.org/browse/JDK-8312489): Increase jdk.jar.maxSignatureFileSize default which is too low for JARs such as WhiteSource/Mend unified agent jar (**Enhancement** - P3)
 * [JDK-8313216](https://bugs.openjdk.org/browse/JDK-8313216): Increase jdk.jar.maxSignatureFileSize default which is too low for JARs such as WhiteSource/Mend unified agent jar (**CSR**)

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)
 * [Matthias Baesken](https://openjdk.org/census#mbaesken) (@MBaesken - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15072/head:pull/15072` \
`$ git checkout pull/15072`

Update a local copy of the PR: \
`$ git checkout pull/15072` \
`$ git pull https://git.openjdk.org/jdk.git pull/15072/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15072`

View PR using the GUI difftool: \
`$ git pr show -t 15072`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15072.diff">https://git.openjdk.org/jdk/pull/15072.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15072#issuecomment-1655907314)

### CODE REVIEW

File: src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java [hunk 1]
```diff
@@ -848,15 +848,15 @@ private static int initializeMaxSigFileSize() {
          * in a JAR file.
          */
         Integer tmp = GetIntegerAction.privilegedGetProperty(
-                "jdk.jar.maxSignatureFileSize", 8000000);
+                "jdk.jar.maxSignatureFileSize", 16000000);
         if (tmp < 0 || tmp > MAX_ARRAY_SIZE) {
```
Comment (mcpowers) [inline src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java:852]: Can't `tmp` be `null`? After all, it is an `Integer` object. The rest of your code changes look fine to me.

Comment (haimaychao) [reply src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java:852]: `tmp` will not be null. `GetIntegerAction.privilegedGetProperty()` will return the default value when jdk.jar.maxSignatureFileSize=null

Comment (mcpowers) [reply src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java:852]: You're right. Didn't look close enough.

File: src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java [hunk 2]
```diff
@@ -848,15 +848,15 @@ private static int initializeMaxSigFileSize() {
...
             if (debug != null) {
-                debug.println("Default signature file size 8000000 bytes " +
+                debug.println("Default signature file size 16000000 bytes " +
                         "is used as the specified size for the " +
                         "jdk.jar.maxSignatureFileSize system property " +
                         "is out of range: " + tmp);
```
Comment (seanjmullan) [inline src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java:857]: The wording of the message is a bit hard to understand. I suggest rewording slightly as: 
```
                debug.println("The default signature file size of 16000000 bytes " +
                        "will be used as the specified size for the " +
                        "jdk.jar.maxSignatureFileSize system property " +
                        "since the value is out of range: " + tmp);
```

**Review by seanjmullan — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java (hunk 1 above)
Comment (cl4es) [reply src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java:852]: Might make sense to declare `tmp` as an `int` to avoid future confusion and save a few bytecode on repeated implicit unboxing every time `tmp` is used.

Comment (haimaychao) [reply src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java:852]: Done as suggested.

File: src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java (hunk 2 above)
Comment (haimaychao) [reply src/java.base/share/classes/sun/security/util/SignatureFileVerifier.java]: Reworded the debug message.

**Review by MBaesken — APPROVED** [review]
Thanks for the adjustment. Looks good to me.

**Comment by FrankReisenhofer** [comment]
Please have a look at [1]

Mend has released a fix 

[1] https://github.com/whitesource/unified-agent-distribution/issues/48

**Review by seanjmullan — APPROVED** [review]
